app/controllers/repositories/show/prefixes/new.js

```diff
@@ -22,7 +24,44 @@ export default Controller.extend({
         'page[size]': 10
       })
       .then(function (providerPrefixes) {
-        self.set('provider-prefixes', providerPrefixes);
+        if (providerPrefixes.length >= 10) {
```

Comment:
Isn't the below the same logic as findPrefix() above? Is it possible to extract this out to avoid duplication.
Unless Im missing a difference.

Reply:
Probably.  I will check.

Reply:
They are very, very similar.  The one here is building a free list of 10 prefixes, drawing from the free provider prefixes, then drawing from the general pool. It sets up that list for the 'assign prefix' function.  

findPrefix in app/controllers/providers/show/repositories/new.js is looking for 1 prefix to assign to a newly created repository.

Probably could abstract something out.  I will spend an hour or two looking at it tomorrow, but I am not sure how much time I should spend looking at it at this point.

Reply:
@richardhallett I extracted out shared code from findPrefix (app/controllers/providers/show/repositories/new.js) and searchPrefix(app/controllers/repositories/show/prefixes/new.js). I put it into a function in the prefixes service (app/services/prefix.js) that gets available prefixes.  It seems to work.